libraries/core_libs/network/src/tarcap/shared_states/peers_state.cpp

```diff
@@ -28,6 +27,22 @@ std::shared_ptr<TaraxaPeer> PeersState::getPendingPeer(const dev::p2p::NodeID& n
   return nullptr;
 }
 
+std::pair<std::shared_ptr<TaraxaPeer>, bool> PeersState::getAnyPeer(const dev::p2p::NodeID& node_id) const {
+  std::shared_lock lock(peers_mutex_);
+
+  auto it_peer = peers_.find(node_id);
+  if (it_peer != peers_.end()) {
+    return {it_peer->second, false};
+  }
+
+  it_peer = pending_peers_.find(node_id);
+  if (it_peer != pending_peers_.end()) {
```

Comment:
I would use this form for both ifs and not reuse it_peer as peers and pending peers are different structures.... 

```
if (auto it_peer = pending_peers_.find(node_id); it_peer != pending_peers_.end()) {
  ...
}
```